operators/multiclusterobservability/controllers/multiclusterobservability/multiclusterobservability_controller.go

```diff
@@ -130,8 +131,9 @@ func (r *MultiClusterObservabilityReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, err
 	}
 
+	// check if the MCH CRD exists
 	mchCrdExists, _ := r.CRDMap[config.MCHCrdName]
-	if req.Name == config.MCHUpdatedRequestName && mchCrdExists {
+	if mchCrdExists {
```

Comment:
how about flow like below?

if mchCrdExists && imageList is empty {
    if req.Name == config.MCHUpdatedRequestName {
        // read mch CR and then read image list
    } else {
        return and reconcile later
    }
}

this flow can avoid duplicate checking on mch, also fix the problem we met before

Reply:
So we will only reconcile the `MCHUpdated` request? 
For the first reconcile, it's OK. But what about the changes of the watching objects(configmap, secret....)?

Reply:
we just need that reconcile to read the image info and store in memory. 
in the following reconciles, they can use the value directly, correct?

Reply:
the only concern is that the image manifest change after that may not be caught.

Reply:
I added ` imageManifestsMap is empty` check to avoid read configmap for each reconcile.
The `if req.Name == config.MCHUpdatedRequestName` is not added because if the MCO is recreated, the operator will not catch the `MCHUpdatedRequest` event.